apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/ServerExecutor.java

```diff
@@ -0,0 +1,285 @@
+          appRunConsumer.getStdoutContents().contains(successMessage));
+    }
+    // do nothing if we've already got a server started
+
+    return true;
+  }
+
+  /**
+   * Create a command for the server launcher script to be run with.
+   *
+   * @param workDirectory the server-work directory
+   * @param dbUrl the db url
+   * @param dbUsername the db username
+   * @param dbPassword the db password
+   * @return the command array for the server app
+   */
+  private static String[] createCommandForServerLauncherApp(
+      String workDirectory, String dbUrl, String dbUsername, String dbPassword) {
+    try {
+      Path assemblyDirectory =
+          Files.list(Paths.get(workDirectory))
+              .filter(f -> f.getFileName().toString().startsWith("bfd-server-launcher-"))
+              .findFirst()
+              .orElse(Path.of(""));
+      Path scriptPath = findFirstPathMatchWithFilenameEnd(assemblyDirectory.toString(), ".sh", 1);
+
+      String gcLog = workDirectory + "/gc.log";
+      String maxHeapArg = System.getProperty("its.bfdServer.jvmargs", "-Xmx4g");
+      String containerImageType =
+          System.getProperty("its.testcontainer.db.image", TEST_CONTAINER_DATABASE_IMAGE_DEFAULT);
+      // FUTURE: Inherit these from system properties? Which of these are valuable to pass?
+      String v2Enabled = "true";
+      String pacEnabled = "true";
+      String pacOldMbiHashEnabled = "true";
+      String pacClaimSourceTypes = "fiss,mcs";
+      String includeFakeDrugCode = "true";
+      String includeFakeOrgName = "true";
+      Random rand = new Random();
+      // Copied this from the startup script, but may not be needed
+      String bfdServerId = String.valueOf(rand.nextInt(10240));
+
+      List<String> args = new ArrayList<>();
+      args.add(scriptPath.toAbsolutePath().toString());
+      args.add(maxHeapArg);
+      args.add(String.format("-Xlog:gc*:%s:time,level,tags", gcLog));
+      args.add(String.format("-Dbfd-server-%s", bfdServerId));
+      args.add(String.format("-DbfdServer.v2.enabled=%s", v2Enabled));
+      args.add(String.format("-DbfdServer.pac.enabled=%s", pacEnabled));
+      args.add(String.format("-DbfdServer.pac.oldMbiHash.enabled=%s", pacOldMbiHashEnabled));
+      args.add(String.format("-DbfdServer.pac.claimSourceTypes=%s", pacClaimSourceTypes));
+      args.add(String.format("-DbfdServer.db.url=%s", dbUrl));
+      args.add(String.format("-DbfdServer.db.username=%s", dbUsername));
+      args.add(String.format("-DbfdServer.db.password=%s", dbPassword));
+      args.add(String.format("-DbfdServer.include.fake.drug.code=%s", includeFakeDrugCode));
+      args.add(String.format("-DbfdServer.include.fake.org.name=%s", includeFakeOrgName));
+      args.add(String.format("-Dits.testcontainer.db.image=%s", containerImageType));
+      return args.toArray(new String[0]);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
This could be removed, the upper level method already throws IOException and it doesnt really matter if we throw that or this; I'll fix this if there are other adjustments to make.